examples/governance/apps/src/bin/publisher.rs

```diff
@@ -0,0 +1,175 @@
+    pub async fn send(&self, calldata: Vec<u8>) -> Result<Option<TransactionReceipt>> {
+        let tx = TransactionRequest::new()
+            .chain_id(self.chain_id)
+            .to(self.contract)
+            .from(self.client.address())
+            .data(calldata);
+
+        log::info!("Transaction request: {:?}", &tx);
+
+        let tx = self.client.send_transaction(tx, None).await?.await?;
+
+        log::info!("Transaction receipt: {:?}", &tx);
+
+        Ok(tx)
+    }
+}
+
+/// Arguments of the publisher CLI.
+#[derive(Parser, Debug)]
+#[clap(author, version, about, long_about = None)]
+struct Args {
+    /// Ethereum chain ID
+    #[clap(long)]
+    chain_id: u64,
+
+    /// Ethereum Wallet Private Key
+    #[clap(long, env)]
+    eth_wallet_private_key: String,
+
+    /// Node RPC URL
+    #[clap(long)]
+    rpc_url: String,
+
+    /// Application's contract address on Ethereum
+    #[clap(long)]
+    contract: String,
+
+    /// The proposal ID (32 bytes, hex-encoded)
+    #[clap(long)]
+    proposal_id: String,
+
+    /// The votes data (hex-encoded, multiple of 100 bytes)
+    #[clap(long)]
+    votes_data: String,
+}
+
+fn main() -> Result<()> {
+    env_logger::init();
+    // Parse CLI Arguments: The application starts by parsing command-line arguments provided by the user.
+    let args = Args::parse();
+
+    // Create a new transaction sender using the parsed arguments.
+    let tx_sender = TxSender::new(
+        args.chain_id,
+        &args.rpc_url,
+        &args.eth_wallet_private_key,
+        &args.contract,
+    )?;
+
+    // Decode the hex-encoded proposal ID and votes data
+    let proposal_id = decode(&args.proposal_id).context("Failed to decode proposal ID")?;
+    let votes_data = decode(&args.votes_data).context("Failed to decode votes data")?;
+
+    // Validate input lengths
+    if proposal_id.len() != 32 {
+        return Err(anyhow::anyhow!("Proposal ID must be 32 bytes"));
+    }
+    if votes_data.len() % 100 != 0 {
+        return Err(anyhow::anyhow!(
+            "Votes data must be a multiple of 100 bytes"
+        ));
+    }
+
+    // Combine proposal ID and votes data
+    let input = [&proposal_id[..], &votes_data[..]].concat();
+    let env = ExecutorEnv::builder().write_slice(&input).build()?;
+
+    let receipt = default_prover()
+        .prove_with_ctx(
+            env,
+            &VerifierContext::default(),
+            FINALIZE_VOTES_ELF,
+            &ProverOpts::groth16(),
+        )?
+        .receipt;
+
+    // Compute the ImageID of `finalize_votes.rs` ELF binary.
+    // let image_id = compute_image_id(FINALIZE_VOTES_ELF)?;
+    // let image_id_bytes: [u8; 32] = image_id
+    //     .as_bytes()
+    //     .try_into()
+    //     .expect("Digest should be 32 bytes");
+    // let image_id_fixed_bytes = FixedBytes::<32>::from(image_id_bytes);
+
+    // Encode the seal with the selector.
+    let seal = groth16::encode(receipt.inner.groth16()?.seal.clone())?;
```

Comment:
```suggestion
    let seal = risc0_ethereum_contracts::encode_seal(&receipt)?;
```

Using this new method allows you to be compatible with `RISC0_DEV_MODE` as well

Reply:
that's much better and makes sense thanks @capossele